mods/default/craftitems.lua

```diff
@@ -112,6 +112,16 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
 		end
 		data.description = "\""..short_title.."\" by "..data.owner
 		data.text = fields.text:sub(1, max_text_size)
+		while(true) do
+			local num_start,num_end = data.text:find("\r")
+			if num_start ~= nil and num_end ~= nil then
+				local str_start = data.text:sub(1,num_start-1)
+				local str_end = data.text:sub(num_end+1)
+				data.text = str_start .. "\n" .. str_end
+			else
+				break
+			end
+		end
```

Comment:
you can simplify everything into `data.text = data.text:gsub("\r", "\n")`

Reply:
@sfan5 that will cause an unintended additional new line for `\r\n` line endings.
`data.text = data.text:gsub("\r\n", "\n")`

Reply:
I know that, what I suggested is just shorter code for what he described.

Reply:
Hmm. This works better.